### ISSUE TITLE
FileNotFoundError: emoji_list.py

### ISSUE THREAD

**Issue by yochananmarqos** [mijorus/smile#3]
I tried building 1.1.5 from source, but the `generate_emoji_dict.py` script failed:

```
Running custom install script '/usr/bin/python /build/smile/src/smile-1.1.5/build-aux/meson/emoji_list/generate_emoji_dict.py'
--- stdout ---

--- stderr ---
Traceback (most recent call last):
  File "/build/smile/src/smile-1.1.5/build-aux/meson/emoji_list/generate_emoji_dict.py", line 72, in <module>
    output_file = open(f"{datadir}/emoji_list.py", 'w+')
FileNotFoundError: [Errno 2] No such file or directory: '/usr/share/smile/smile/assets/emoji_list.py'

FAILED: install script '/usr/bin/python /build/smile/src/smile-1.1.5/build-aux/meson/emoji_list/generate_emoji_dict.py' exit code 1, stopped
```

**Comment by mijorus** [comment]
Are you trying to built it directly with meson, or are you building the Flatpak?

**Comment by mijorus** [comment]
> I'm building it with Meson.

As this is my first Linux app, I just targeted flatpak amd didn't test the building process outside of it.

However, that seems to be a problem with python not creating the file on the first run.

Could you please share the python version/os that you are using so I can spin up a VM and possibly push a fix?

Thank you

**Comment by yochananmarqos** [comment]
I'm using Manjaro (unstable branch) with Python 3.10.2 and Meson 0.61.1. Here's my [PKGBUILD](https://github.com/yochananmarqos/pkgbuilds/blob/master/smile/PKGBUILD).

**Comment by mijorus** [comment]
I was working on a fix but then I remembered that this app heavily relies on gnomes's `org.gnome.Sdk:41` runtime in order to display emojis correctly. 

This is because some of the latest emojis (like skin tones or flags) are displayed as two different emojis on unsupported systems, like [this one](https://emojipedia.org/kiss-woman-man/) for example.

This would break the symmetry of the table and would look awful.
In other words, unless i re-write the app to use some built-in images Smile is unusable outside of a flatpak.

**Comment by yochananmarqos** [comment]
That's not the point. The files should be available to Meson, not referencing something you already have installed in your system during build.

**Comment by mijorus** [comment]
> That's not the point. The files should be available to Meson, not referencing something you already have installed in your system during build.

Which version of python are you using? In the error, it looks like it complaints about a file not found, while it should be creating it (note the 'w+' syntax).

**Comment by yochananmarqos** [comment]
I'm using Python 3.10.4.

**Comment by yochananmarqos** [comment]
The file structure doesn't look right in `/usr/share/smile/`. The `meson.build` should not be installed and there should not be a `/usr/share/smile/smile/` directory:

```
.
└── usr
    └── share
        └── smile
            ├── assets
            │   ├── meson.build
            │   └── NotoColorEmoji.ttf
            ├── smile
            │   ├── AboutDialog.py
            │   ├── assets
            │   │   ├── smile.autostart.desktop
            │   │   └── style.css
            │   ├── CustomTagEntry.py
            │   ├── __init__.py
            │   ├── lib
            │   │   ├── custom_tags.py
            │   │   ├── emoji_history.py
            │   │   └── user_config.py
            │   ├── main.py
            │   ├── meson.build
            │   ├── Picker.py
            │   ├── Settings.py
            │   ├── ShortcutsWindow.py
            │   ├── smile.gresource
            │   ├── smile.gresource.xml
            │   ├── smile.in
            │   ├── ui
            │   │   ├── menu.xml
            │   │   ├── settings.glade
            │   │   ├── settings.glade~
            │   │   └── shortcuts.xml
            │   └── utils.py
            └── smile.gresource

```

**Comment by eli-schwartz** [comment]
> Which version of python are you using? In the error, it looks like it complaints about a file not found, while it should be creating it (note the 'w+' syntax).

Python reports "File Not Found" when it cannot find the directory the created file needs to be created in. So, the directory `/usr/share/smile/smile/assets/` does not exist.

This makes sense, since https://mesonbuild.com/Reference-manual_builtin_meson.html#mesonadd_install_script documents that you should use `MESON_INSTALL_DESTDIR_PREFIX` but you are using `MESON_INSTALL_PREFIX` instead.

The actual file that is supposed to be getting created is:

```
/build/smile/pkg/smile/usr/share/smile/smile/assets/emoji_list.py
```

**Comment by yochananmarqos** [comment]
@eli-schwartz Aha! That does the trick.

**Comment by mijorus** [comment]
lol I am so noob with this.

Btw I fully recognize that "building" a python file is not actually the cleanest thing ever, but hey that does the trick

**Comment by eli-schwartz** [comment]
Well, you might want to create it using custom_target() instead.

Independent of how you create it, it's probably not completely ideal to close the ticket so quickly.
